### ISSUE TITLE
[Fuzzer] Escaped trim function call?

### ISSUE THREAD

**Issue by Tishj** [duckdb/duckdb#4586]
```
DuckDB db(nullptr);
Connection con(db);
con.EnableQueryVerification();
con.SendQuery("SELECT \"trim\"(1);");
```

Actual:
src/main/client_context.cpp:931: std::string duckdb::ClientContext::VerifyQuery(duckdb::ClientContextLock &, const std::string &, unique_ptrduckdb::SQLStatement): Assertion `orig_expr_list[i]->Equals(verify_statements[v_idx].select_list[i].get())' failed

Expected:
The query should output a single 1 value.

**Comment by Tishj** [comment]
Creating a test like this:
```cpp
TEST_CASE("Test escaped function call", "[api]") {
  DuckDB db(nullptr);
  Connection con(db);
  con.EnableQueryVerification();
  con.SendQuery("SELECT \"trim\"(1);");
}
```

Does not crash, so I'm marking this as fixed as it's not reproducable

**Comment by Tishj** [comment]
The issue is that the parser looks for the specific name "trim"

```yak
			| TRIM '(' BOTH trim_list ')'
				{
					/* various trim expressions are defined in SQL
					 * - thomas 1997-07-19
					 */
					$$ = (PGNode *) makeFuncCall(SystemFuncName("trim"), $4, @1);
				}
```
Which then gets transformed to have the schema "main" (DEFAULT_SCHEMA)
which it won't do when the string is `\"trim\"`

**Comment by Mytherin** [comment]
Fixed by #4552
